include/mxnet/ndarray.h

```diff
@@ -414,6 +431,8 @@ class NDArray {
   size_t offset_;
   /*! \brief type of data */
   int dtype_ = -1;
+
+  nnvm::NodeEntry entry_{nnvm::Node::Create(), 0, 0};
```

Comment:
Create a Node instance then cast it as a shared pointer (https://github.com/dmlc/nnvm/blob/master/src/core/node.cc#L40), should I profile this?